src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -53,12 +53,12 @@ public DateTime StartTime
             get
             {
                 // Get the RUsage data and convert the process start time (which is the number of
-                // nanoseconds before Now that the process started) to a DateTime.
-                DateTime now = DateTime.UtcNow;
+                // nanoseconds elapse from boot to that the process started) to a DateTime.
                 Interop.libproc.rusage_info_v3 info = Interop.libproc.proc_pid_rusage(_processId);
-                int milliseconds = Convert.ToInt32(info.ri_proc_start_abstime / MillisecondFactor);
-                TimeSpan ts = new TimeSpan(0, 0, 0, 0, milliseconds);
-                return now.Subtract(ts).ToLocalTime();
+                double milliseconds = info.ri_proc_start_abstime / MillisecondFactor;
+                TimeSpan ts = TimeSpan.FromMilliseconds(milliseconds);
+                DateTime uptime = DateTime.UtcNow.Subtract(TimeSpan.FromMilliseconds(Environment.TickCount));
+                return uptime.Add(ts).ToLocalTime();
```

Comment:
It seems like both the Linux and OS X code are now getting a time-since-boot value in an OS-specific way but then both doing the same post-processing on it to get the start time... can we unify that post-processing into a single helper that's shared between them?